### ISSUE TITLE
Black frame at the end of each clip

### ISSUE THREAD

**Issue by Floutche** [OpenShot/openshot-qt#938]
Why does "open shot" add a black frame at the end of each clip when I drop them on the timeline
can that feature be customized ?

**Windows 7 professionnel*
* OpenShot Version: 2.4.0

**Comment by gjinxers** [comment]
Floutche -- I am also using Windows 7, and OpenShot is not adding any black frames at the end of my clips when I add them into my tracks on the timeline.

Just a question for you..... Are you overlapping your clips to create automatic transitions or adding in transitions between the clips manually????   When you don't overlap or add transitions, a black box appears in your preview window because that space on the track in the timeline is empty. Black seems to be the default video preview color of the blank track space on the timeline.

Hope this helps. My apologies if it doesn't.
Good luck with your issue though.

**Comment by DylanC** [comment]
@Floutche - You must have gaps between your clips. This feature cannot be customized. I suggest you move your clips closer together or add a different colour image between the clips.

**Comment by singalen** [comment]
I witness this on 2.4.1/Mac.
I made my best effort to snap the clips on the timeline together, but there are still black gaps, visible in OpenShot.
The clips are imported from iPhone SE.
I can provide samples.

**Comment by DylanC** [comment]
@singalen - Go ahead. Thanks.

**Comment by singalen** [comment]
Emailed urls to photoflare info box.

**Comment by DylanC** [comment]
@singalen - Thanks. I will try this tomorrow or soon.

**Comment by Floutche** [comment]
I can try it too. But I don't understand what to do. Could you explain it to me ?

**Comment by DylanC** [comment]
@singalen - I played each clip in a video player and there seems to be a black frame before the video starts in each one. Can you try them in a video player to confirm?

**Comment by singalen** [comment]
mpv 0.27/Mac, no black frame either at the beginning or the end.

```
$ mpv -v IMG_0142.MOV
[cplayer] Command line options: '-v' 'IMG_0142.MOV'
[cplayer] mpv 0.27.0 (C) 2000-2017 mpv/MPlayer/mplayer2 projects
[cplayer]  built on Sat Sep 23 20:53:45 CEST 2017
[cplayer] ffmpeg library versions:
[cplayer]    libavutil       55.58.100
[cplayer]    libavcodec      57.89.100
[cplayer]    libavformat     57.71.100
[cplayer]    libswscale      4.6.100
[cplayer]    libavfilter     6.82.100
[cplayer]    libswresample   2.7.100
[cplayer] ffmpeg version: 3.3.4
[cplayer]
[cplayer] Configuration: waf configure --prefix=/Users/djinn/apps/homebrew/Cellar/mpv/0.27.0 --enable-zsh-comp --enable-libmpv-shared --enable-html-build --confdir=/Users/djinn/apps/homebrew/etc/mpv --datadir=/Users/djinn/apps/homebrew/Cellar/mpv/0.27.0/share/mpv --mandir=/Users/djinn/apps/homebrew/Cellar/mpv/0.27.0/share/man --docdir=/Users/djinn/apps/homebrew/Cellar/mpv/0.27.0/share/doc/mpv --zshdir=/Users/djinn/apps/homebrew/Cellar/mpv/0.27.0/share/zsh/site-functions
[cplayer] List of enabled features: 52 apple-remote asm atomics avutil-imgcpy-uc avutil-spherical bsd-fstatfs build-date cocoa coreaudio cplayer cplugins cuda-hwaccel debug-build encoding fchmod gl gl-cocoa glob glob-posix gnuc iconv is_ffmpeg jpeg lcms2 libass libass-osd libav libavcodec libavdevice libdl libm libmpv-shared lua nanosleep optimize osx-thread-name plain-gl posix posix-or-mingw posix-spawn pthreads shm stdatomic termios videotoolbox-gl videotoolbox-hwaccel videotoolbox-hwaccel-old zlib zsh-comp

...

[cplayer]  (+) Video --vid=1 (*) (h264 1920x1080 29.977fps)
[cplayer]  (+) Audio --aid=1 --alang=und (*) (aac 1ch 44100Hz)

...

[vd] Container reported FPS: 29.976881
[vd] Codec list:
[vd]     h264 - H.264 / AVC / MPEG-4 AVC / MPEG-4 part 10
[vd]     h264_vda (h264) - H.264 (VDA acceleration)
[vd] Opening video decoder h264
[vd] Using software decoding.
[vd] Detected 8 logical cores.
[vd] Requesting 9 threads for decoding.
[vd] Selected video codec: h264 (H.264 / AVC / MPEG-4 AVC / MPEG-4 part 10)
```

What codec do OpenShot and your player use?

**Comment by DylanC** [comment]
@singalen - I was using the Totem media player on Ubuntu 17.10. I think it uses mmpeg 3.3.4 too. I will do a trace similar to yours and add it here. Also I could try those clips on another OS to see if any difference.

**Comment by N3WWN** [comment]
@DylanC This may be related to https://github.com/OpenShot/openshot-qt/issues/733 where I reported a blank frame at the start of exported videos.  I hadn't noticed one at the end, but I wasn't looking for one, either. 😀

**Comment by DylanC** [comment]
@N3WWN - It was actually the source clips in this case. He send them to me via e-mail. When I played them just in a video player I saw a black frame at the start of the clips.

I will try the source clips on another OS just to be sure.

**Comment by singalen** [comment]
Wait, wait, wait. If three other players (mpv, ffplay, QuickTime) don't show a black frame in the source clips, it was NOT the source, but rather the video engine. 

Totem's engine is GStreamer, IIRC.
Does OpenShot also use GStreamer?

Can you run Totem/gst-launch with higher GST_DEBUG levels and see if there's any diagnostics in these videos?

**Comment by DylanC** [comment]
@singalen - Yes, it does sound like a video engine problem. Totem does use GStreamer. OpenShot uses mmpeg as far as I know.

I'll run some more tests before commenting back here.

**Comment by machra1961** [comment]
I also get this problem importing from a Garmin dashcam. I am using Linux Mint as the OS and when I view the files in the default media player they are fine.  However, once they are imported into a project each clip has 3 or 4 black frames at the end of the clip.  To be exact the last frame is a 'good' frame but is preceded by 3 or 4 black frames.  This is a real pain because I am trying to stitch multiple clips together and if I just add them to the timeline having imported them I get black spaces in the finished clip.  What I have to do is to cut the last 5 frames off the end of EACH clip before stitching them together on the timeline.  Fine if stitching a few clips but a real nuisance when I am tryin gto stitch together 80x 1 min clips (journey by car on holiday I wanted to keep).

**Comment by peanutbutterandcrackers** [comment]
Does the issue still persist in version 2.4.2?

**Comment by singalen** [comment]
Didn't reproduce on 2.4.2.

**Comment by peanutbutterandcrackers** [comment]
Great! need more confirmation now! But yay!

**Comment by DylanC** [comment]
@peanutbutterandcrackers I'm happy enough with this confirmation. Hope you don't mind if I close. Others can let us know later on.

**Comment by peanutbutterandcrackers** [comment]
It's all right, Cap'ain! We can re-open it later if it is necessary. But I think it's good. :+1:

**Comment by machra1961** [comment]
I have just checked and black frames have gone.  However, the last frame of the video is now repeated 3x times more at the end ie you get a moving frame and then 3x more exact replicas of that last moving frame, and as a rsult when such a video is stitched to another one you get a jumpy transition. Putting it into a frames context I get frames - 1,2,3,3,3,3,4,5,6 when I stich two videos together where frames 1,2&3 are from the first video and fromes 4,5&6 from the second.

**Comment by jonathon-oodar** [comment]
@machra1961 did you open a new issue for this? Noticing the same problem

**Comment by machra1961** [comment]
Hi. No I didn’t. I haven’t used Openshot for a while but from what I remember it did get fixed in a later update. However, I do seem to remember that maybe not fully, and as you are having the same problem obviously it didn’t. Real shame as it is the best video editor I have used, but this did make it unusable for what I was doing.

**Comment by peanutbutterandcrackers** [comment]
I thought the issue was fixed in 2.4.3, with the patch that had got into libopenshot. If this issue still persists, I suggest you open a new issue - and give all the details. Details about the media file you are using, openshot version, etc. and also please do not forget to attach the logs. That should help the devs debug this soon enough.
